pkg/deploy/stack_deployer.go

```diff
@@ -54,35 +60,140 @@ type latticeServiceStackDeployer struct {
 	svcBuilder            gateway.LatticeServiceBuilder
 }
 
+var tgGcOnce sync.Once
+var tgGc *TgGc
+
 func NewLatticeServiceStackDeploy(
 	log gwlog.Logger,
 	cloud pkg_aws.Cloud,
 	k8sClient client.Client,
 ) *latticeServiceStackDeployer {
 	brTgBuilder := gateway.NewBackendRefTargetGroupBuilder(log, k8sClient)
 
+	tgMgr := lattice.NewTargetGroupManager(log, cloud)
+	tgSvcExpBuilder := gateway.NewSvcExportTargetGroupBuilder(log, k8sClient)
+	svcBuilder := gateway.NewLatticeServiceBuilder(log, k8sClient, brTgBuilder)
+
+	tgGcOnce.Do(func() {
+		// TODO: need to refactor TG synthesizer. Remove stack from constructor
```

Comment:
Is this TODO a future issue?

Reply:
We should revisit synthesizers, the concept of abstract stack that we build and how it's actually implemented is questionable.